flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -99,6 +99,18 @@ public static void setupYarnClassPath(Configuration conf, Map<String, String> ap
 		}
 	}
 
+	/**
+	 * Returns the Path where the YARN application files should be uploaded to.
+	 *
+	 * @param homedir
+	 * 		remote home directory base (will be extended)
+	 * @param appId
+	 * 		application ID
+	 */
+	public static Path getYarnFilesDir(final Path homedir, final String appId) {
```

Comment:
I think we can tighten the contract a bit and say that `appId` must be of type `ApplicationId`.